### DISCUSSION TITLE
API usage tutorial

### DISCUSSION THREAD

**Discussion by cgx** [pixelfed/pixelfed#5235] [Q&A] [answered]
Hi! I'm trying to query my private pixelfed instance through the API, but without much success.

I've created a personal access token to authenticate, but calling any non-public API endpoint returns a 403 with an empty "error" message.

Here's an example:
```
curl -X GET \
∙ --url 'https://pixelfed.local/api/v1/notifications' \
∙ -H 'accept: application/json' \
∙ -H 'Authorization: Bearer <personal_access_token>'
```
And the response:
```
{"error":""}
```

Am I missing a header in my query? Is there any parameter that must be added to my server configuration?

Thanks!

**Comment by dansup** [comment]
That looks correct, I'm able to successfully query my notifications on pixelfed.social using the example on https://beta-preview.pixelfed.io/resources/notifications#retrieve-your-notifications and my PAT

How did you generate your PAT?

**Reply by cgx** [reply]
My token looks valid since I get a *Unauthenticated* error message if I try with a random string.

I've debug a bit more and found out that the request aborts on `$request→user()→tokenCan('read')`; the `scopes` attribute of my token is actually an empty array. How is this array normally populated?

Thanks!

**Reply by cgx** [reply]
Is it possible that I should send some commands to Passport to make it work?

Thanks!

**Reply by cgx** [reply, marked as answer]
I solved my problem by manually editing the column `scopes` for my PAT in the `oauth_access_tokens` table.
